### PR TITLE
Added logging of CAN servo and ESC status

### PR DESCRIPTION
**Pull request by tridge**
i didn't use the existing ESCn msgs as I wanted to use instance logging, plus there are some slightly different fields. We could unify them in the future

### CODE REVIEW

**Comment by tridge** [comment]
This has been successfully tested on a heli with Hitec servos

**Comment by tridge** [comment]
we've tested this on zubax ESCs for ESC logging works nicely

**Comment by JoshWelsh** [comment]
Flew this today, works as expected.  Very exciting!

File: libraries/AP_Logger/LogStructure.h [hunk 1]
```diff
@@ -951,6 +951,28 @@ struct PACKED log_Esc {
...
+    float force;
+    float speed;
+    uint8_t power_pct;
+};
+
+struct PACKED log_CESC {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;     
+    uint8_t id;
+    uint32_t error_count;
```
Comment (peterbarker) [inline libraries/AP_Logger/LogStructure.h]: Do we really need to burn this many bytes?  I'm assuming this is a high-rate message.

File: libraries/AP_Logger/LogStructure.h [hunk 2]
```diff
@@ -951,6 +951,28 @@ struct PACKED log_Esc {
...
+    float voltage;
+    float current;
+    float temperature;
+    int32_t rpm;
```
Comment (peterbarker) [inline libraries/AP_Logger/LogStructure.h]: Again - int32_t for RPM?  Sadly, 32k RPM is actually not completely improbable.

**Comment by VDLJu** [comment]
Hi,

@tridge
Please check if there could be issues with ESC logging. I got corrupted logs on 4.0.1-rc2 with UAVCAN ESC's powered (Zubax Myxa):

https://discuss.ardupilot.org/t/copter-4-0-1-rc2-available-for-beta-testing-critical-fix-included/50940/32?u=vdlju

https://discuss.ardupilot.org/t/copter-4-0-1-rc2-available-for-beta-testing-critical-fix-included/50940/33?u=vdlju

**Pull request merged**